src/main/java/duke/ui/UserInterface.java

```diff
@@ -0,0 +1,62 @@
+package duke.ui;
+import java.io.InputStream;
+import java.io.PrintStream;
+import java.util.Scanner;
+import java.util.stream.Collectors;
+
+import duke.tasks.TaskList;
+
+public class UserInterface {
+    private InputStream in;
+    private PrintStream out;
+    private Scanner sc;
+
+    public UserInterface() {
+        in = System.in;
+        out = System.out;
+        sc = new Scanner(in);
+    }
+
+    private void printLine() {
+        out.println("____________________________________________________________");
+    }
+
+    public void showGreeting() {
+        printLine();
+        out.println();
+        out.println("Hello! I'm Duke");
+        out.println("What can I do for you?");
+        printLine();
+    }
+
+    public void showExitMessage() {
```

Comment:
Perhaps it'd be a good idea to close the scanner over here. Makes the code more robust.